### PR TITLE
signature: checks for integer overflow in limits propagation

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/3840

Describe changes:
- Checks for integer overflows in limits propagation

There may be over integer overflows in this function (unsigned ones not triggering UBSAN)
We may want to range check distance to -65535:65535

### CODE REVIEW

File: src/detect-content.c [hunk 1]
```diff
@@ -553,7 +553,11 @@ void DetectContentPropagateLimits(Signature *s)
                             (cd->flags & (DETECT_CONTENT_DEPTH|DETECT_CONTENT_OFFSET|DETECT_CONTENT_WITHIN|DETECT_CONTENT_DISTANCE)) == (DETECT_CONTENT_DISTANCE)) {
                         if (cd->distance >= 0) {
                             // only distance
-                            offset = cd->offset = offset_plus_pat + cd->distance;
+                            if ((uint32_t)offset_plus_pat + cd->distance <= UINT16_MAX) {
```
Comment (jlucovsky) [inline src/detect-content.c:556]: nit: suggest parens around the operands for clarity.

Even though `<=` is lower precedence, it may be clearer.

Comment (catenacyber) [reply src/detect-content.c:556]: Do you mean `((uint32_t)offset_plus_pat) + ` ?

**Comment by victorjulien** [comment]
Merged in #5258, thanks!

**Comment by catenacyber** [comment]
There seem to be other integer overflows in this function (unsigned ones not triggering UBSAN)
Should I create a redmine ticket to check for them ?

**Pull request closed without merging**